### PR TITLE
AIP-84: Migrating GET queued asset events for assets to FASTAPI

### PR DESCRIPTION
**Pull request by vatsrahul1001**
Related: https://github.com/apache/airflow/issues/42370

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: airflow/api_fastapi/core_api/routes/public/assets.py [hunk 1]
```diff
@@ -252,3 +252,47 @@ def get_dag_asset_queued_events(
...
+        QueuedEventResponse(created_at=adrq.created_at, dag_id=adrq.target_dag_id, uri=uri)
+        for adrq, uri in adrqs
+    ]
+
+    return QueuedEventCollectionResponse(
+        queued_events=[
+            QueuedEventResponse.model_validate(queued_event, from_attributes=True)
+            for queued_event in queued_events
+        ],
+        total_entries=total_entries,
```
Comment (pierrejeambrun) [inline airflow/api_fastapi/core_api/routes/public/assets.py]:
```suggestion
    queued_events = [
        QueuedEventResponse(created_at=adrq.created_at, dag_id=adrq.target_dag_id, uri=uri)
        for adrq, uri in adrqs
    ]

    return QueuedEventCollectionResponse(
        queued_events=queued_events,
        total_entries=total_entries,
```

File: airflow/api_fastapi/core_api/routes/public/assets.py [hunk 2]
```diff
@@ -252,3 +252,47 @@ def get_dag_asset_queued_events(
         ],
         total_entries=total_entries,
     )
+
+
+@assets_router.get(
+    "/assets/queuedEvent/{uri:path}",
```
Comment (pierrejeambrun) [inline airflow/api_fastapi/core_api/routes/public/assets.py]:
```suggestion
    "/assets/queuedEvents/{uri:path}",
```

**Review by pierrejeambrun — COMMENTED** [review]
Nice a couple of small suggestions, should be good to merge.

File: airflow/api_fastapi/core_api/routes/public/assets.py [hunk 3]
```diff
@@ -276,3 +276,44 @@ def delete_dag_asset_queued_events(
 
     if result.rowcount == 0:
         raise HTTPException(status.HTTP_404_NOT_FOUND, f"Queue event with dag_id: `{dag_id}` was not found")
+
+
+@assets_router.get(
+    "/assets/queuedEvents/{uri:path}",
```
Comment (pierrejeambrun) [inline airflow/api_fastapi/core_api/routes/public/assets.py]:
```suggestion
    "/assets/queuedEvent/{uri:path}",
```

This change will be done in a follow up PR on all `queueEvent` endpoints to keep consistency.

Comment (vatsrahul1001) [reply airflow/api_fastapi/core_api/routes/public/assets.py]: @pierrejeambrun is there an issue for that I can pick that up?

Comment (vatsrahul1001) [reply airflow/api_fastapi/core_api/routes/public/assets.py]: Done

**Review by pierrejeambrun — APPROVED** [review]
Nice

**Pull request merged**